subsys/bluetooth/host/gatt.c

```diff
@@ -5531,6 +5530,10 @@ void bt_gatt_encrypt_change(struct bt_conn *conn)
 	data.conn = conn;
 	data.sec = BT_SECURITY_L1;
 
+#if defined(BT_GATT_AUTO_RESUBSCRIBE)
+	add_subscriptions(conn);
+#endif
```

Comment:
Since `add_subscriptions` is a static function, we could remove the guard from `add_subscriptions` and use `IS_ENABLED` here; the linker should still optimize the function out.